src/datagrid/DataGridFilter.tsx

```diff
@@ -147,17 +147,31 @@ export class DataGridFilter extends React.PureComponent<DataGridFilterProps, Dat
         const {isOpen} = this.state;
 
         if (isOpen) {
-            // Calculate left and top for filter box
-            const filterBoxTop = this.refParent.current!.getClientRects()[0].top + 15;
-            const filterBoxLeft =
-                this.refParent.current!.getClientRects()[0].left -
-                this.refChild.current!.getClientRects()[0].width +
-                20;
-            const transformVal = "translateX(" + filterBoxLeft + "px) " + "translateY(" + filterBoxTop + "px)";
-            this.setState({transformVal: transformVal});
+            let prevChildWidth = this.refChild.current!.getClientRects()[0].width;
+            if (this.state!.transformVal === "translateX(0px) translateY(0px)") {
+                this.updateFitlerPosition(prevChildWidth);
+            } else {
+                const handleResize = setInterval(() => {
+                    let nextChildWidth = this.refChild.current!.getClientRects()[0].width;
```

Comment:
Should we check if these things exist ?  Check if its necessary.

```suggestion
                    let nextChildWidth = this.refChild && this.refChild.current && this.refChild.current.getClientRects()[0] && this.refChild.current.getClientRects()[0].width;
```